api/src/main/java/org/pmiops/workbench/db/dao/WorkspaceServiceImpl.java

```diff
@@ -126,26 +135,33 @@ public void updateUserRoles(String ns, String id, Set<WorkspaceUserRole> userRol
     }
 
     for(WorkspaceUserRole currentWorkspaceUser : dbWorkspace.getWorkspaceUserRoles()) {
-      if(currentWorkspaceUser.getUser().getFreeTierBillingProjectName() == null) {
-        throw new ServerErrorException("User not yet initialized in FireCloud.");
-      }
       WorkspaceACLUpdate currentUpdate = new WorkspaceACLUpdate();
       currentUpdate.setEmail(currentWorkspaceUser.getUser().getEmail());
       currentUpdate.setCanCompute(false);
       if (currentWorkspaceUser.getRole() == WorkspaceAccessLevel.OWNER) {
         currentUpdate.setCanShare(true);
-        currentUpdate.setAccessLevel("OWNER");
+        currentUpdate.setAccessLevel(WorkspaceAccessLevel.OWNER.toString());
       } else if (currentWorkspaceUser.getRole() == WorkspaceAccessLevel.WRITER) {
         currentUpdate.setCanShare(false);
-        currentUpdate.setAccessLevel("WRITER");
+        currentUpdate.setAccessLevel(WorkspaceAccessLevel.WRITER.toString());
       } else {
         currentUpdate.setCanShare(false);
-        currentUpdate.setAccessLevel("READER");
+        currentUpdate.setAccessLevel(WorkspaceAccessLevel.READER.toString());
       }
       updateACLRequestList.add(currentUpdate);
     }
     try {
-      fireCloudService.updateWorkspaceACL(ns, id, false, updateACLRequestList);
+      WorkspaceACLUpdateResponseList fireCloudResponse = fireCloudService.updateWorkspaceACL(ns, id, false, updateACLRequestList);
+      if (fireCloudResponse.getUsersNotFound().size() != 0) {
+        String usersNotFound = "";
+        for (int i = 0; i < fireCloudResponse.getUsersNotFound().size(); i++) {
+          if (i > 0) {
+            usersNotFound += ", ";
+          }
+          usersNotFound += fireCloudResponse.getUsersNotFound().get(i).getEmail();
+        }
+        throw new ServerErrorException("Could not find users: " + usersNotFound);
```

Comment:
This shouldn't be ServerErrorException; that's for 500s / bugs in our code / unhandled conditions. This is a case of a bad request... use BadRequestException.

This will allow you to pass in an ErrorResponse where you could list out each user that can't be found... which is better than just having a string like this that the client would have a harder time parsing (if it wanted to, say, flag the particular users that don't exist in the UI.)